c7n/resources/iam.py

```diff
@@ -82,7 +83,7 @@ class resource_type(object):
         service = 'iam'
         type = 'role'
         enum_spec = ('list_roles', 'Roles', None)
-        detail_spec = None
+        detail_spec = ('get_role', 'RoleName', 'RoleName', 'Role')
```

Comment:
unfortunately this line is going to cause a bunch of test churn

Reply:
i've gone ahead and pulled this one line change and its accompany test churn into a separate pull request that is now a pre-requisite for this in #3588